internal/pkg/comm/server.go

```diff
@@ -245,6 +252,8 @@ func (gServer *GRPCServer) AppendClientRootCAs(clientRoots [][]byte) error {
 
 // internal function to add a PEM-encoded clientRootCA
 func (gServer *GRPCServer) appendClientRootCA(clientRoot []byte) error {
+	gServer.tls.lock.Lock()
```

Comment:
this works, but is very brittle as no one promises you that the TLS config object is not going to be modified further in the code execution.